### PR TITLE
Document usage of `gum` crate

### PR DESCRIPTION
**Pull request by mrcnski**
# PULL REQUEST

## Overview

Add some basic usage info to `gum` crate.

### CODE REVIEW

File: node/gum/src/lib.rs [hunk 1]
```diff
@@ -20,6 +20,88 @@
...
+//!     "failed to send a handshake to the spawned worker",
+//! );
+//! ```
+//!
+//! ### Log levels
+//!
+//! Instead of `warn!` you can use one of the other available macros, which are
+//! based on the [`tracing`
+//! macros](https://docs.rs/tracing/latest/tracing/index.html#macros). In decreasing
+//! order of priority they are:
```
Comment (sandreim) [inline node/gum/src/lib.rs:46]:
```suggestion
//!  All of the the [`tracing` macros](https://docs.rs/tracing/latest/tracing/index.html#macros) log level macros are available.
//! In decreasing order of priority they are:
```

File: node/gum/src/lib.rs [hunk 2]
```diff
@@ -20,6 +20,88 @@
...
+//! - `info!`
+//! - `debug!`
+//! - `trace!`
+//!
+//! ### `target`
+//!
+//! The `LOG_TARGET` should be defined once per crate, e.g.:
+//!
+//! ```rs
+//! const LOG_TARGET: &str = "parachain::pvf::prepare-worker";
```
[the remark this answers is not in the document]
Comment (sandreim) [reply node/gum/src/lib.rs:59]: This string is not arbitrary, it produces the above set of labels which are applied to each log entry. These are used when debugging by specializing the Grafana Loki query to filter specific subsystem logs. The more specific the query is the better when approaching the query response limit (5000 lines in our setup).

An example of such a query: `{node=~"versi-validator-a-node-6", target="parachain", subtarget="approval-voting"} ` . This returns all the approval voting logs, and reduces the query duration time, by only looking at logs for the parachain consensus target `parachain`. 


 We should reflect this in the usage.

[the remark this answers is not in the document]
Comment (mrcnski) [reply node/gum/src/lib.rs:57]: Okay I see the target and subtarget in your query, will update!

[the remark this answers is not in the document]
Comment (sandreim) [reply node/gum/src/lib.rs:57]: I don't think you'd find anything in the code, as this is an opaque string. The magic should happen in Grafana Loki, and our setup currently has this configuration.

[the remark this answers is not in the document]
Comment (sandreim) [reply node/gum/src/lib.rs]: https://grafana.parity-mgmt.parity.io/goto/j1zBa5QVg?orgId=1

[the remark this answers is not in the document]
Comment (mrcnski) [reply node/gum/src/lib.rs]: I didn't know that, thanks for the info! Updated.

**Review by sandreim — APPROVED** [review]

**Review by alexggh — APPROVED** [review]

**Comment by sandreim** [comment]
bot merge

**Pull request merged**